packages/arb-token-bridge-ui/src/util/TokenUtils.ts

```diff
@@ -235,3 +238,49 @@ export async function l1TokenIsDisabled({
   const erc20Bridger = await Erc20Bridger.fromProvider(l2Provider)
   return erc20Bridger.l1TokenIsDisabled(erc20L1Address, l1Provider)
 }
+
+type SanitizeTokenOptions = {
+  erc20L1Address?: string | null // token address on L1
+  chain: Chain // chain for which we want to retrieve the token name / symbol
+}
+
+// get the exact token symbol for a particular chain
+export function sanitizeTokenSymbol(
+  tokenSymbol: string,
+  options?: SanitizeTokenOptions
+) {
+  if (typeof options === 'undefined') {
+    return tokenSymbol.toUpperCase()
+  }
+
+  const isArbitrumOne = isNetwork(options.chain.id).isArbitrumOne
+
+  // only special case for USDC is Arbitrum One
+  if (options.erc20L1Address === CommonAddress.Mainnet.USDC && isArbitrumOne) {
+    return 'USDC.e'
+  }
+
+  return tokenSymbol.toUpperCase()
+}
+
+// get the exact token name for a particular chain
+export function sanitizeTokenName(
+  tokenName: string,
+  options?: SanitizeTokenOptions
+) {
+  if (typeof options === 'undefined') {
+    return tokenName
+  }
+
+  const isArbitrumOne = isNetwork(options.chain.id).isArbitrumOne
+  const isMainnet = isNetwork(options.chain.id).isMainnet
+
+  // only special case for USDC is Arbitrum One / Mainnet
+  if (options.erc20L1Address === CommonAddress.Mainnet.USDC) {
```

Comment:
```suggestion
  if (options.erc20L1Address.toLowerCase() === CommonAddress.Mainnet.USDC.toLowerCase()) {
```

maybe it can be a `isMainnetUSDC()` function

Reply:
Hmmm, it's definitely nicer, though idk how it will scale if we have like 10+ addresses on multiple chains

Reply:
Done